jobs: avoid unnecessary writes

stevendanna opened the pull request:
See individual commits.

stevendanna committed:
jobs: avoid unnecessary system.job_progress updates

This avoid unnecessary job progress updates when a user issues job
control statements such as pause and unpause.

Epic: none
Release note: None

stevendanna committed:
jobs: avoid duplicate system.jobs updates

This avoids cases where we issue 2 updates to system.jobs instead of 1.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152658)
<!-- Reviewable:end -->

stevendanna commented:
@dt I haven't looked at these closely, so perhaps the change is non-sense. But I noted these when reading this code yesterday.

stevendanna commented:
The test failures _seem_ like good things:

```
    rtt_analysis_bench.go:224: Jobs/resume_job: got 9, expected 12-15
```

stevendanna commented:
Rebased.

Clause says:

```
> Any bugs?

....

  No bugs found in the job system commits
```

Your milage may vary.

stevendanna committed:
crdb_interal: always report non-null fraction fraction_completed

We have code that assumes it can always scan this into a float, which
isn't the case when it is NULL.

Epic: none
Release note: None

stevendanna commented:
Claude lied.
